src/store/sagas/visualization-saga.test.js

```diff
@@ -68,128 +82,141 @@ describe('visualization-saga', () => {
     })
   })
 
-
-  describe('createVisualization', () => {
-    const id = "hello"
-    const title = "my first visualization"
-    const query = "select hello from world"
-    const returnedVisualization = { id, title, query }
-    const initialState = {
-      visualization: { chart: { data: { x: [1, 2, 3], xsrc: "col1" }, layout: {}, frames: [] } },
-      queryReducer: { queryData: [{ col1: 1 }, { col1: 2 }, { col1: 3 }] }
-    }
-
-    describe('successfully', () => {
-      var dispatched = []
-      beforeEach(async () => {
-        AuthenticatedHTTPClient.post = jest.fn(() => ({ status: 200, data: returnedVisualization }))
-        dispatched = await recordSaga(saveVisualization, visualizationSave(title, query), initialState)
-      })
-
-      it('calls api with parameters that include a dereferenced chart', () => {
-        expect(AuthenticatedHTTPClient.post).toHaveBeenCalledWith(`/api/v1/visualization`,
-          {
-            title: title,
-            query: query,
-            chart: { data: { x: [], xsrc: "col1" }, frames: [], layout: {} } // TODO: Double check the expected x
-          }
-        )
-      });
-
-      it('signals the visualization is available', () => {
-        expect(dispatched).toContainEqual(visualizationSaveSuccess(returnedVisualization))
+  describe('saveVisualization', () => {
+    describe('without an id', () => {
+      const title = "my first visualization"
+      const query = "select hello from world"
+      const returnedVisualization = { id: "generated id from api", title, query }
+      const initialState = {
+        visualization: { chart: { data: { x: [1, 2, 3], xsrc: "col1" }, layout: {}, frames: [] } },
+        queryReducer: { queryData: [{ col1: 1 }, { col1: 2 }, { col1: 3 }] }
+      }
+
+      describe('successfully', () => {
+        var dispatched = []
+        beforeEach(async () => {
+          AuthenticatedHTTPClient.post = jest.fn(() => ({ status: 200, data: returnedVisualization }))
+          dispatched = await recordSaga(saveVisualization, visualizationSave({title, query}), initialState)
+        })
+
+        it('calls api with parameters that include a dereferenced chart', () => {
+          expect(AuthenticatedHTTPClient.post).toHaveBeenCalledWith(`/api/v1/visualization`,
+            {
+              title: title,
+              query: query,
+              chart: { data: { x: [], xsrc: "col1" }, frames: [], layout: {} } // TODO: Double check the expected x
```

Comment:
Is this comment temporary? (Aka going to live forever?)

Reply:
Nope... we didn't add it - probably just the result of a refactor.  We'll remove it.  Thanks.